### PR TITLE
Adds rare chance for surgery trays to become medical toolboxes

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Surgery trays now have a small chance to become medical toolboxes. Autopsy trays can become coroner toolboxes.

image

They work basically the same, just that they can't be rolled around and don't slow you down when walking around.

Medical toolboxes are fairly weak, but coroner toolboxes are pretty strong.

Added a 1 in 1.000.000 chance for a toolbox to have four latches.

## Why It's Good For The Game

> Surgery trays now have a small chance to become medical toolboxes. Autopsy trays can become coroner toolboxes.
> They work basically the same, just that they can't be rolled around and don't slow you down when walking around.

The idea of going to the surgery room and finding a _medical toolbox_ instead of a surgery tray is inexplicably amusing to me.

> Medical toolboxes are fairly weak, but coroner toolboxes are pretty strong.

Healers and hurters!

> Added a 1 in 1.000.000 chance for a toolbox to have four latches.

peak absurdity is reached

~~next pr will include brown toolboxes, which clip to your belt~~

credit 2 @SmArtKar for sprites

## Changelog

:cl:
carlarc, smartkar
add: Surgery trays now have a small chance to become medical toolboxes. Autopsy trays can become coroner toolboxes.
add: Added a 1 in 1.000.000 chance for a toolbox to have four latches.
/:cl:

### CODE REVIEW

File: _maps/shuttles/whiteship_birdshot.dmm [hunk 1]
```diff
@@ -146,7 +146,7 @@
 	pixel_y = 3
 	},
 /obj/item/reagent_containers/blood,
-/obj/item/surgery_tray/full{
+/obj/effect/spawner/surgery_tray/full{
```
Comment (Jacquerel) [inline _maps/shuttles/whiteship_birdshot.dmm:149]: Is it really necessary for this to be the case on every whiteship and shuttle, as well as on the station? 
It's fine for some places to actually map in the item.

File: code/game/objects/items/storage/toolbox.dm [hunk 2]
```diff
@@ -225,6 +227,49 @@
...
+	icon_state = "medical"
+	inhand_icon_state = "medical_toolbox"
+	attack_verb_continuous = list("treats", "surgeries", "tends", "tends wounds on")
+	attack_verb_simple = list("treat", "surgery", "tend", "tend wounds on")
+	w_class = WEIGHT_CLASS_BULKY
+	material_flags = NONE
+	force = 5 // its for healing
+	wound_bonus = 25 // wounds are medical right?
+	///var/latches = "single_latch"
+	///var/has_latches = TRUE
```
Comment (Jacquerel) [inline code/game/objects/items/storage/toolbox.dm:242]: You forgot to delete these

**Review by Wallemations — APPROVED** [review]
lol

File: code/game/objects/items/surgery_tray.dm [hunk 3]
```diff
@@ -157,59 +168,89 @@
 
 /obj/item/surgery_tray/deployed
 	is_portable = FALSE
+	trace_if_mapped_in = FALSE
```
Comment (Jacquerel) [inline code/game/objects/items/surgery_tray.dm:171]: I think this is unecessarily protective. Mappers should be able to choose to map in a tray (or a toolbox) if they so desire.

File: code/game/objects/items/surgery_tray.dm [hunk 4]
```diff
@@ -157,59 +168,89 @@
...
+	)
+	trace_if_mapped_in = FALSE
+
+/obj/effect/spawner/surgery_tray
+	name = "surgery tray spawner"
+	icon = 'icons/obj/medical/medicart.dmi'
+	icon_state = "tray"
+	var/tray_to_spawn = /obj/item/surgery_tray
+	var/rare_toolbox_replacement = /obj/item/storage/toolbox/medical
+	var/toolbox_chance = 1
```
Comment (Jacquerel) [inline code/game/objects/items/surgery_tray.dm:236]: Missing comments

**Comment by FlufflesTheDog** [comment]
> Added a 1 in 1.000.000 chance for a toolbox to have four latches.

guarantee that anyone that finds one is going to have no idea how rare it is lmao
I have found maybe 2 triple-latch toolboxes and not a single person I showed had any idea what was special about them

I love it

**Comment by necromanceranne** [comment]
Could you do me a favour and give the coroner toolbox undead bane?

**Comment by Jackal-boop** [comment]
Sprite bl- *my mouth is covered and I am dragged into the shadows*

File: _maps/shuttles/whiteship_birdshot.dmm (hunk 1 above)
Comment (carlarctg) [reply _maps/shuttles/whiteship_birdshot.dmm:149]: it's not necessary nor unnecessary. it just is

File: code/game/objects/items/surgery_tray.dm (hunk 3 above)
Comment (carlarctg) [reply code/game/objects/items/surgery_tray.dm:171]: ok

**Comment by mc-oofert** [comment]
we already have medical toolboxes we just call them surgery trays
this is essentially an 11-slot toolbox that you may put anything in and without the slowdown of the surgery tray

**Comment by necromanceranne** [comment]
I feel like you have missed the bit entirely jimmy.

**Comment by mc-oofert** [comment]
> I feel like you have missed the bit entirely jimmy.

if the bit is that its a toolbox and its literally just a tray without slowdown then its not a good bit

**Comment by Loafin34** [comment]
i would give it the same whitelist of surgical tools as the tray. I don't think a funny duffel bag is on the menu

**Comment by carlarctg** [comment]
> > I feel like you have missed the bit entirely jimmy.
> 
> if the bit is that its a toolbox and its literally just being able to get a tray without slowdown and item restrictions in a 1% chance and add stupid spawners to maps for something that at best rolls 3 (presuming 3 trays) in 100 rounds then its not a good bit

🤓

**Comment by Yobrocharlie** [comment]
> > I feel like you have missed the bit entirely jimmy.
> 
> if the bit is that its a toolbox and its literally just being able to get a tray without slowdown and item restrictions in a 1% chance and add stupid spawners to maps for something that at best rolls 3 (presuming 3 trays) in 100 rounds then its not a good bit

You don’t sound like a good bit, you sound stinky go take a shower

File: code/game/objects/items/storage/toolbox.dm [hunk 5]
```diff
@@ -225,6 +227,51 @@
...
+	w_class = WEIGHT_CLASS_BULKY
+	material_flags = NONE
+	force = 5 // its for healing
+	wound_bonus = 25 // wounds are medical right?
+	/// Tray we steal the og contents from.
+	var/obj/item/surgery_tray/tray_type = /obj/item/surgery_tray
+
+/obj/item/storage/toolbox/medical/Initialize(mapload)
+	. = ..()
+	// what do any of these numbers fucking mean
```
Comment (Xander3359) [inline code/game/objects/items/storage/toolbox.dm:246]: Hey so please don't just leave shitpost comments.

Comment (carlarctg) [reply code/game/objects/items/storage/toolbox.dm:246]: image

**Review by Ghommie — APPROVED** [review]
I wonder if this works with SmartKar's toobox PR but I guess it does (i don't remember if toolboxes can hold surgery tools rn).

**Pull request merged**